### PR TITLE
fix: courses without lectures

### PR DESCRIPTION
**Pull request by dhenkel92**
This can happen if a user deletes all lectures after the course has been created. The problem is that the course is not editable anymore and the user-app just crashes.

This PR will make sure that this doesn't happen anymore.

https://lern-fair.slack.com/archives/C03E45MES4X/p1689785056516519

### CODE REVIEW

File: src/pages/course-creation/CourseAppointments.tsx [hunk 1]
```diff
@@ -58,7 +58,6 @@ const CourseAppointments: React.FC<Props> = ({ next, back, isEditing, appointmen
         return convertedAppointments;
     };
     const canGoFurther = () => {
-        if (isEditing) return appointments.length === 0 ? true : false;
```
Comment (dhenkel92) [inline src/pages/course-creation/CourseAppointments.tsx:61]: I have no idea what was the idea behind this, but it prevents us from editing a course that doesn't have lectures.

Comment (Jonasdoubleyou) [reply src/pages/course-creation/CourseAppointments.tsx:61]: Hm, that state should not be reachable. All courses should at least have one lecture.

**Review by Jonasdoubleyou — APPROVED** [review]

File: src/pages/course-creation/CourseAppointments.tsx (hunk 1 above)
Comment (dhenkel92) [reply src/pages/course-creation/CourseAppointments.tsx:61]: But then we also have to communicate to the user how to replace all lectures.

Comment (Jonasdoubleyou) [reply src/pages/course-creation/CourseAppointments.tsx:61]: Fair, the UI for course creation / editing was always messy, I wanted to revisit that problem once we have appointments. I guess if we allow this again then somewhen we will have the opposite bugs in the backend, were we rely on this assumption. But fine for me if it fixes the immediate problem

Comment (Jonasdoubleyou) [reply src/pages/course-creation/CourseAppointments.tsx:61]: Hm, at least https://github.com/corona-school/backend/blob/master/common/courses/states.ts looks as if we at least handle this in the majority of cases

Comment (dhenkel92) [reply src/pages/course-creation/CourseAppointments.tsx:61]: That's true. Prisma helps a lot here.

**Pull request merged**